front/src/modules/pathfinding/components/IncompatibleConstraints/IncompatibleConstrainstFilters.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useCallback, type FC, type HTMLAttributes } from 'react';
+
+import { Filter } from '@osrd-project/ui-icons';
+import cx from 'classnames';
+import { toPairs } from 'lodash';
+import { useTranslation } from 'react-i18next';
+
+import type { IncompatibleConstraintTypes } from './type';
+import { getIcon } from './utils';
+
+interface IncompatibleConstraintsFiltersProps extends HTMLAttributes<unknown> {
+  data: Record<IncompatibleConstraintTypes, { count: number; enabled: boolean }>;
```

Comment:
To replace by the state type you created

```suggestion
  incompatibleConstraintsDictionnary: StateType;
```